src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Telemetry/DesignTimeTelemetryLogger.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.Build.Framework;
+
+namespace Microsoft.VisualStudio.Telemetry
+{
+    internal sealed class DesignTimeTelemetryLogger : ILogger
+    {
+        private sealed class TargetRecord
+        {
+            public TargetRecord(string targetName, DateTime started)
+            {
+                TargetName = targetName;
+                Started = started;
+                Ended = DateTime.MinValue;
+            }
+
+            public string TargetName { get; }
+
+            public DateTime Started { get; }
+
+            public DateTime Ended { get; set; }
+
+            public TimeSpan Elapsed => Ended - Started;
+        }
+
+        private readonly ITelemetryService _telemetryService;
+
+        private readonly Dictionary<int, TargetRecord> _targets = new Dictionary<int, TargetRecord>();
+        private bool _succeeded;
+
+        public DesignTimeTelemetryLogger(ITelemetryService telemetryService)
+        {
+            _telemetryService = telemetryService;
+        }
+
+        public LoggerVerbosity Verbosity { get; set; }
+
+        public string Parameters { get; set; }
+
+        public void Initialize(IEventSource eventSource)
+        {
+            eventSource.TargetStarted += TargetStarted;
+            eventSource.TargetFinished += TargetFinished;
+            eventSource.BuildFinished += BuildFinished;
+        }
+
+        private void BuildFinished(object sender, BuildFinishedEventArgs e)
+        {
+            _succeeded = e.Succeeded;
+        }
+
+        private void TargetStarted(object sender, TargetStartedEventArgs e)
+        {
+            _targets[e.BuildEventContext.TargetId] = new TargetRecord(e.TargetName, e.Timestamp);
```

Comment:
This needs to be handle overlapping targets.

Reply:
CallTarget/MSBuild can cause targets to be nested.

Reply:
But the target IDs are unique so there won't be overwriting, correct? Or are you saying that because targets can be nested, an outer target will be attributed the time it spent in nested targets as well? That's unfortunate but that's a bigger scope of work, I'm just trying to get the rough extent at this point.

Reply:
kk.